plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMigrateProcedure.java

```diff
@@ -224,20 +224,115 @@ public void testMigrateUnsupportedColumnType()
     {
         String tableName = "test_migrate_unsupported_column_type_" + randomNameSuffix();
         String hiveTableName = "hive.tpch." + tableName;
+
+        assertUpdate("CREATE TABLE " + hiveTableName + " AS SELECT tinyint '1' x", 1);
+        assertQuery("SELECT * FROM " + hiveTableName, "SELECT 1");
+
+        assertQueryFails("CALL iceberg.system.migrate('tpch', '%s')".formatted(tableName), "Type not supported for Iceberg: tinyint");
+
+        assertQuery("SELECT * FROM " + hiveTableName, "SELECT 1");
+
+        assertUpdate("DROP TABLE " + hiveTableName);
+    }
+
+    @DataProvider
+    public static Object[][] timestampPrecisionsAndValuesOrc()
+    {
+        return new Object[][] {
+                {HiveTimestampPrecision.MILLISECONDS, "2021-01-01 06:12:12.123"},
+                {HiveTimestampPrecision.MICROSECONDS, "2021-01-01 06:12:12.123456"}
+        };
```

Comment:
I tried adding 

```
{HiveTimestampPrecision.NANOSECONDS, "2021-01-01 06:12:12.123456789"},
```

and stumbled on the message


```
Actual rows (up to 100 of 1 extra rows shown, 1 rows in total):
    [2021-01-01T06:12:12.123457]
Expected rows (up to 100 of 1 missing rows shown, 1 rows in total):
    [2021-01-01T06:12:12.123456789]
```


The value from the source table is different than the value from the migrated table.
However `migrate` silently performed the migration.
I'm not sure whether this should be the expected scenario.

Reply:
Hive timestamp has precision 9, nanoseconds. Iceberg timestamp has precision 6, microseconds.
So the only choice we have is

1. fail migration if table has any timestamp columns
2. do migration and timestamp will be read in Iceberg rounded (OR truncated)
3. do migration and but fail the reads later, if timestamp values are not full microseconds.

From user perspective, I think that `migrate` is often viewed as a more performant and more convenient (in-place) replacement to CTAS, and with CTAS we want behavior (2), so I'd go with (2).

Reply:
I added the nanoseconds case, timestamp are rounded to precision 6.